<a href="https://colab.research.google.com/github/nihdin/AppliedEconometrics/blob/master/AEA7.ipynb" target="_parent"><img src="https://colab.research.google.com/assets/colab-badge.svg" alt="Open In Colab"/></a>

W16.4


For the second condition, the rank condition will not hold. Comparing the first and second equations, there are no exogenous variables that appear in the first equation, that are not present in the second equation. When the value of ϒ3 is different from zero (as stated in the question), the equation 1 can be identified. We would presume the value of ϒ3 to be less than zero by intuition. This will lead us to get an exogenous variable, log(price) that can be used as an Instrumental variable for alcohol in estimating the first equation by 2SLS

In [3]:
install.packages("wooldridge")
library(wooldridge)
install.packages("systemfit")
library(systemfit)

Installing package into ‘/usr/local/lib/R/site-library’
(as ‘lib’ is unspecified)

Installing package into ‘/usr/local/lib/R/site-library’
(as ‘lib’ is unspecified)

Loading required package: Matrix

Loading required package: car

Loading required package: carData

Loading required package: lmtest

Loading required package: zoo


Attaching package: ‘zoo’


The following objects are masked from ‘package:base’:

    as.Date, as.Date.numeric



Please cite the 'systemfit' package as:
Arne Henningsen and Jeff D. Hamann (2007). systemfit: A Package for Estimating Systems of Simultaneous Equations in R. Journal of Statistical Software 23(4), 1-40. http://www.jstatsoft.org/v23/i04/.

If you have questions, suggestions, or comments regarding the 'systemfit' package, please use a forum or 'tracker' at systemfit's R-Forge site:
https://r-forge.r-project.org/projects/systemfit/



In [0]:
attach(smoke)

In [5]:
age2<-age*age
m1<-lm(log(income)~cigs+educ+age+ age2)
m1
summary(m1)


Call:
lm(formula = log(income) ~ cigs + educ + age + age2)

Coefficients:
(Intercept)         cigs         educ          age         age2  
  7.7954433    0.0017306    0.0603606    0.0576908   -0.0006306  



Call:
lm(formula = log(income) ~ cigs + educ + age + age2)

Residuals:
    Min      1Q  Median      3Q     Max 
-3.6237 -0.2978  0.1314  0.4167  1.3542 

Coefficients:
              Estimate Std. Error t value Pr(>|t|)    
(Intercept)  7.795e+00  1.704e-01  45.741  < 2e-16 ***
cigs         1.731e-03  1.714e-03   1.010    0.313    
educ         6.036e-02  7.898e-03   7.642 6.10e-14 ***
age          5.769e-02  7.644e-03   7.548 1.21e-13 ***
age2        -6.306e-04  8.338e-05  -7.563 1.08e-13 ***
---
Signif. codes:  0 ‘***’ 0.001 ‘**’ 0.01 ‘*’ 0.05 ‘.’ 0.1 ‘ ’ 1

Residual standard error: 0.6529 on 802 degrees of freedom
Multiple R-squared:  0.165,	Adjusted R-squared:  0.1608 
F-statistic: 39.61 on 4 and 802 DF,  p-value: < 2.2e-16


Assuming the causal relationship, we interpret $\beta_1$ as follows. 100*$\beta_1$ is the approximate change in the income if the individual smokes an extra cigarette per day.

Since consumption and price are negatively related with everything else remaining constant, we do expect that $\gamma_5$ to be less than or equal to zero. In a similar manner, we can understand that the restaurant smoking restrictions should reduce cigarette smoking and therefor $\gamma_6$ should also be less than or equal to zero.

The incom equation from part (i) is identified when we have $\gamma_5$ or $\gamma_6$ is different from the value zero. In other words, we need one exogenous variable that is not in the log(income) equation.

#####log (**_$income$_**) = 7.80 + 0.0017 **_$cigs$_** + 0.060 **_$educ$_** + 0.058 **_$age$_** - 0.00063 **_$age^{2}$_**

**_$n$_**= 807
**_$R^{2}$_**= 0.165

The coefficient on **_$cigs$_** (**_$\beta_1$_**) implies that smoking cigarattes will cause income to increase, even though the coefficient is not statistically different from zero.

In [6]:
m2<-lm(cigs~educ+age+age2+log(cigpric)+restaurn)
m2
summary(m2)


Call:
lm(formula = cigs ~ educ + age + age2 + log(cigpric) + restaurn)

Coefficients:
 (Intercept)          educ           age          age2  log(cigpric)  
     1.58013      -0.45015       0.82254      -0.00959      -0.35132  
    restaurn  
    -2.73639  



Call:
lm(formula = cigs ~ educ + age + age2 + log(cigpric) + restaurn)

Residuals:
    Min      1Q  Median      3Q     Max 
-15.078  -9.312  -6.246   8.038  70.329 

Coefficients:
              Estimate Std. Error t value Pr(>|t|)    
(Intercept)   1.580127  23.695583   0.067  0.94685    
educ         -0.450147   0.161640  -2.785  0.00548 ** 
age           0.822541   0.154322   5.330 1.28e-07 ***
age2         -0.009590   0.001679  -5.711 1.58e-08 ***
log(cigpric) -0.351320   5.765550  -0.061  0.95143    
restaurn     -2.736389   1.109693  -2.466  0.01388 *  
---
Signif. codes:  0 ‘***’ 0.001 ‘**’ 0.01 ‘*’ 0.05 ‘.’ 0.1 ‘ ’ 1

Residual standard error: 13.41 on 801 degrees of freedom
Multiple R-squared:  0.051,	Adjusted R-squared:  0.04508 
F-statistic:  8.61 on 5 and 801 DF,  p-value: 5.859e-08


#####**_$cigs$_** = 1.5801 - 0.4501 **_$educ$_** + 0.8225 **_$age$_** - 0.0096 **_$age^{2}$_** - 0.3513 **_$log(cigpric$_**) -2.7364 **_$restaurn$_**

**_$n$_**= 807
**_$R^{2}$_**= 0.051

log(**_$cigpric$_**) is not statistically significant. But **_$restaurn$_** has a t statitsic of -2.466 and a negative sign. **_$restaurn$_** is statitsically signicant at a p-value less than 0.05

In [7]:
k1=log(income)~cigs+educ+age+ age2
k2=cigs ~ educ + age + age2 + log(cigpric) + restaurn
sys= list(k1, k2)
iv= ~educ+age+age2+lcigpric+restaurn
fit.sys= systemfit(sys, inst=iv, method="2SLS")
summary(fit.sys)


systemfit results 
method: 2SLS 

          N   DF    SSR detRCov   OLS-R2 McElroy-R2
system 1614 1603 144635 76.1474 0.049477   0.332442

      N  DF        SSR        MSE      RMSE        R2    Adj R2
eq1 807 802    621.013   0.774331  0.879961 -0.516902 -0.524467
eq2 807 801 144013.531 179.792173 13.408660  0.051005  0.045081

The covariance matrix of the residuals
         eq1       eq2
eq1 0.774331   7.94174
eq2 7.941741 179.79217

The correlations of the residuals
         eq1      eq2
eq1 1.000000 0.673081
eq2 0.673081 1.000000


2SLS estimates for 'eq1' (equation 1)
Model Formula: log(income) ~ cigs + educ + age + age2
Instruments: ~educ + age + age2 + lcigpric + restaurn

                Estimate   Std. Error  t value   Pr(>|t|)    
(Intercept)  7.780893233  0.229867343 33.84949 < 2.22e-16 ***
cigs        -0.042125739  0.026218445 -1.60672 0.10850910    
educ         0.039674613  0.016281067  2.43686 0.01503207 *  
age          0.093818217  0.023853373  3.93312 9.1106e-05 ***

#####log (**_$income$_**) = 7.78 - 0.042 **_$cigs$_** + 0.040 **_$educ$_** + 0.094 **_$age$_** - 0.00105 **_$age^{2}$_**

**_$n$_** =807

The coefficient on **_$cigs$_** is less than zero (negative) and not statistically significant at 5% (It is almost significant at the 10%) against a two sided alternative. The estimated effect is large and can be interpreted as that each cigarette that is adittionally smoken lowers the predicted income by about 4.2%


By assuming that the state level cigarette prices and restaurant smoking restrictions are exogenous in the income equation is a cause of problem. Incomes can vary by region as restaurant smoking restrictions. In places where income is lower, the restrictions in smoking in restaurants are less likely (controlling for education and age)

In [0]:
attach(cement)
n1<-lm(gprc~gcem+gprcpet+feb+mar+apr+may+jun+jul+aug+sep+oct+nov+dec)
n1
summary(n1)
n2<-lm(gcem~gdefs+gprcpet+feb+mar+apr+may+jun+jul+aug+sep+oct+nov+dec)
n2
summary(n2)
n3<-lm(gcem~gres+gnon+gprcpet+feb+mar+apr+may+jun+jul+aug+sep+oct+nov+dec)
n3
summary(n3)
#n4<-lm(gprc~gcem+gprcpet+feb+mar+apr+may+jun+jul+aug+sep+oct+nov+dec)
#n4
#summary(n4)

When we estimate by OLS, we can see the result as above. The coefficient of $gcem$ and $gprcpet$ are statistically significant. The estimated supply curve slopes down. The coefficient of gcem is negative while that of gprcet is positive.

We would need $gdefs$ to have a non zero coefficiet in the reduced form for $gcem$. As you can see from model n2, the coefficient of gdefs is -1.0541. The p value is not significant and therefore we cannot reject the null hypothesis, and we can conclude that $gdefs$ is not a useful IV for gcem, eventhough it is exogeneous.


From model n3, we can find out that the coefficient of $gres$ is 0.1361, and that of $gnon$ is 1.15. The $gnon$ is statistically significant and therefore we can go ahead with the Instrument Variable estimation

####$gprc$= 0.228 - 0.0106$gcem$ + 0.0605 $gprcpet$ + ...
$n$= 298 $R^{2}$=0.36 

From model for 2SLS, the coefficient on $gcem$ is negative and is only about one-fourth the size of the OLS coefficient and is insignificant. We can conclude that the static supply function for cement can be horizontal with gprc on the vertical axis. 



In [43]:
airfare1<-subset(airfare, year==1997)
attach(airfare1)
summary(airfare1)

      year            id            dist            passen          fare      
 Min.   :1997   Min.   :   1   Min.   :  95.0   Min.   :  27   Min.   : 37.0  
 1st Qu.:1997   1st Qu.: 288   1st Qu.: 505.0   1st Qu.: 209   1st Qu.:116.0  
 Median :1997   Median : 575   Median : 861.0   Median : 340   Median :160.0  
 Mean   :1997   Mean   : 575   Mean   : 989.7   Mean   : 601   Mean   :173.8  
 3rd Qu.:1997   3rd Qu.: 862   3rd Qu.:1304.0   3rd Qu.: 674   3rd Qu.:225.0  
 Max.   :1997   Max.   :1149   Max.   :2724.0   Max.   :7637   Max.   :460.0  
    bmktshr           ldist            y98         y99         y00   
 Min.   :0.1920   Min.   :4.554   Min.   :0   Min.   :0   Min.   :0  
 1st Qu.:0.4587   1st Qu.:6.225   1st Qu.:0   1st Qu.:0   1st Qu.:0  
 Median :0.6156   Median :6.758   Median :0   Median :0   Median :0  
 Mean   :0.6125   Mean   :6.696   Mean   :0   Mean   :0   Mean   :0  
 3rd Qu.:0.7579   3rd Qu.:7.173   3rd Qu.:0   3rd Qu.:0   3rd Qu.:0  
 Max.   :1.0000   Max.   :7

In [44]:
ld2<-((log(dist))^2)
p1<-lm(log(passen)~log(fare)+log(dist)+ld2)
p1
summary(p1)
Ols_4=lm(lpassen~lfare+ldist+ldistsq, data=airfare1)
summary(Ols_4)
reg4= lm(lfare~ldist+ldistsq+concen, data=airfare1)
summary(reg4)
model7= lpassen~lfare+ldist+ldistsq
model8= lfare~ldistsq+ldistsq+concen
sys4=list(model7, model8)
iv4= ~ldist+ldistsq+concen
fit.sys4= systemfit(sys4, inst=iv4, method="2SLS", data=airfare1)
summary(fit.sys4)


Call:
lm(formula = log(passen) ~ log(fare) + log(dist) + ld2)

Coefficients:
(Intercept)    log(fare)    log(dist)          ld2  
    13.2296      -0.3912      -1.5705       0.1158  



Call:
lm(formula = log(passen) ~ log(fare) + log(dist) + ld2)

Residuals:
    Min      1Q  Median      3Q     Max 
-2.8175 -0.5859 -0.1259  0.5307  3.1557 

Coefficients:
            Estimate Std. Error t value Pr(>|t|)    
(Intercept) 13.22961    2.10015   6.299 4.25e-10 ***
log(fare)   -0.39117    0.06725  -5.817 7.78e-09 ***
log(dist)   -1.57048    0.62852  -2.499   0.0126 *  
ld2          0.11576    0.04772   2.426   0.0154 *  
---
Signif. codes:  0 ‘***’ 0.001 ‘**’ 0.01 ‘*’ 0.05 ‘.’ 0.1 ‘ ’ 1

Residual standard error: 0.824 on 1145 degrees of freedom
Multiple R-squared:  0.05674,	Adjusted R-squared:  0.05426 
F-statistic: 22.96 on 3 and 1145 DF,  p-value: 1.961e-14



Call:
lm(formula = lpassen ~ lfare + ldist + ldistsq, data = airfare1)

Residuals:
    Min      1Q  Median      3Q     Max 
-2.8175 -0.5859 -0.1259  0.5307  3.1557 

Coefficients:
            Estimate Std. Error t value Pr(>|t|)    
(Intercept) 13.22961    2.10015   6.299 4.25e-10 ***
lfare       -0.39117    0.06725  -5.817 7.78e-09 ***
ldist       -1.57048    0.62852  -2.499   0.0126 *  
ldistsq      0.11576    0.04772   2.426   0.0154 *  
---
Signif. codes:  0 ‘***’ 0.001 ‘**’ 0.01 ‘*’ 0.05 ‘.’ 0.1 ‘ ’ 1

Residual standard error: 0.824 on 1145 degrees of freedom
Multiple R-squared:  0.05674,	Adjusted R-squared:  0.05426 
F-statistic: 22.96 on 3 and 1145 DF,  p-value: 1.961e-14



Call:
lm(formula = lfare ~ ldist + ldistsq + concen, data = airfare1)

Residuals:
     Min       1Q   Median       3Q      Max 
-1.30251 -0.26202 -0.03003  0.25920  0.99541 

Coefficients:
            Estimate Std. Error t value Pr(>|t|)    
(Intercept)  6.19005    0.88988   6.956 5.87e-12 ***
ldist       -0.93607    0.27184  -3.443 0.000595 ***
ldistsq      0.10807    0.02062   5.240 1.91e-07 ***
concen       0.39504    0.06272   6.299 4.27e-10 ***
---
Signif. codes:  0 ‘***’ 0.001 ‘**’ 0.01 ‘*’ 0.05 ‘.’ 0.1 ‘ ’ 1

Residual standard error: 0.356 on 1145 degrees of freedom
Multiple R-squared:  0.4076,	Adjusted R-squared:  0.406 
F-statistic: 262.6 on 3 and 1145 DF,  p-value: < 2.2e-16



systemfit results 
method: 2SLS 

          N   DF     SSR  detRCov   OLS-R2 McElroy-R2
system 2298 2291 1016.09 0.086625 0.049659   0.364487

       N   DF     SSR      MSE     RMSE        R2    Adj R2
eq1 1149 1145 869.476 0.759368 0.871417 -0.054897 -0.057660
eq2 1149 1146 146.618 0.127939 0.357686  0.401460  0.400415

The covariance matrix of the residuals
         eq1      eq2
eq1 0.759368 0.102607
eq2 0.102607 0.127939

The correlations of the residuals
         eq1      eq2
eq1 1.000000 0.329192
eq2 0.329192 1.000000


2SLS estimates for 'eq1' (equation 1)
Model Formula: lpassen ~ lfare + ldist + ldistsq
Instruments: ~ldist + ldistsq + concen

              Estimate Std. Error  t value   Pr(>|t|)    
(Intercept) 18.0137490  3.2224815  5.59002 2.8367e-08 ***
lfare       -1.1739978  0.3886211 -3.02093  0.0025759 ** 
ldist       -2.1756654  0.7273403 -2.99126  0.0028377 ** 
ldistsq      0.1870288  0.0612885  3.05161  0.0023286 ** 
---
Signif. codes:  0 ‘***’ 0.001 ‘**’ 0.01 ‘*’ 0.

The sign of $\alpha_1$ is negative, since the demand function is downward sloping. As the price increases, the quantity demanded for airtravel decreases

The estimated price elasticity is -0.391 (t statistic= -5.817) (coefficient of log(fare))

We assume that the passenger demand depends only on the airfare and so once we control for price, the passengers are indifferent about the fraction of travel that is accounted for by the largest carrier

#####$log(fare)$ = 6.19 - 0.936 $log(dist)$ + 0.108 $[log(dist)]^2$ + 0.395 $concen$

$n$= 1149, $R^{2}$= 0.408
The coefficient on concen shows a pretty strong link between the concentration and fare. If the concen increases by 0.10, the fare increases by almost 3.95%, t-statistic being 6.3


By using concen as an Instrumental variable for log(fare) the estimated price elasticity is -1.17 which will show a much greater price elasticity than the OLS estimmate. The Instrumental variable estimate will suggest that a one percent increase in fare will lead to a slight decrease of more than 1% drop in passenger demand. The standard error of IV estimate is larger but statistically significant. It has a value of 0.389 compared with the OLS standard error of 0.067.

The coefficient of ldist is about -2.176 and that of ldistsq is 0.187. There will be a transition point which will transfer the influence from the distance on the seats number from negative to positive. That is by mathematical analysis, 2.176/.187 is approximately 5.82, which is a little less than 340 miles. That is at a certain distance, it will have a negative effect on the number of passsengers, and after that distance will have a positive effect. In short, a U relation.


